### PR TITLE
perf: use map and await generateSwagger

### PR DESCRIPTION
**Pull request by retrogtx**
- Uses Map for improved performance in groupAndSortPathsByFirstTag in the sorting function
- Await generateSwagger to ensure complete documentation before server start
- remove an unused import `TagObject`

### CODE REVIEW

**Review by Praashh — APPROVED** [review]
great, this looks good to me, approving the PR. Thank you @retrogtx 🙏.

File: apps/api/v2/src/main.ts [hunk 1]
```diff
@@ -69,32 +68,37 @@ function isOperationObject(obj: any): obj is OperationObject {
...
-    Object.assign(sortedPaths, groupedPaths[tag]);
-  });
+  Array.from(groupedPaths.keys())
+    .sort(customTagSort)
+    .forEach((tag) => {
+      const tagPaths = groupedPaths.get(tag);
+      if (tagPaths) {
+        Object.assign(sortedPaths, tagPaths);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (retrogtx) [reply apps/api/v2/src/main.ts:101]: I wrote it such keeping in my mind it's time complexity. ,maps are better when dealing with dynamic collections of key value pairs. the approach I went with has a time complexity of o(m log m). 

there is also a check `if (tagPaths)`, so we can possibly get rid of lines 80-81 as well.

[the remark this answers is not in the document]
Comment (Praashh) [reply apps/api/v2/src/main.ts:101]: do you have any comparison to show the difference between the previous approach's results and the current one's result ?

[the remark this answers is not in the document]
Comment (retrogtx) [reply apps/api/v2/src/main.ts:101]: i don't think a benchmark is needed as the time complexity is known here, the approach I went with is a few milliseconds slower but it is better for dynamic allocations. if by comparison you mean some benchmark, i havent done it.

**Comment by anikdhabal** [comment]
Thanks for your work @retrogtx , but it's okay now

**Pull request closed without merging**